### DISCUSSION TITLE
Using @apply in plugin

### DISCUSSION THREAD

**Discussion by saadeghi** [tailwindlabs/tailwindcss#3013] [Help]
Is there a way to use `@apply` when adding a Tailwind CSS plugin?  
I'm thinking of something like:  
```
addComponents({
  '.btn': {
    '@apply': 'bg-red-200 rounded p-4 m-1',
  },
})
```

**Comment by simonswiss** [comment]
Hey!

I don't think that's really supported, but you might be interested in this discussion on the topic: https://github.com/tailwindlabs/tailwindcss/discussions/2049

**Reply by saadeghi** [reply]
Thanks @simonswiss  
I ended up using this solution:  
- I used `@apply` in my .css files
- convert them to real css code using `postcss-cli` and Tailwind
- convert them to css-in-js using `prejss-cli`
- `require` the js files in my Tailwind plugin

[This](https://github.com/saadeghi/daisyui) is the result if anyone's interested
